src/opm/parser/eclipse/EclipseState/Aquifer/NumericalAquifer/SingleNumericalAquifer.cpp

```diff
@@ -52,4 +57,107 @@ namespace Opm {
     size_t SingleNumericalAquifer::numConnections() const {
         return this->connections_.size();
     }
+
+    std::array<std::set<size_t>, 3> SingleNumericalAquifer::transToRemove(const EclipseGrid& grid) const {
+        std::array<std::set<size_t>, 3> trans;
+        for (const auto& cell : this->cells_) {
+            const size_t i = cell.I;
+            const size_t j = cell.J;
+            const size_t k = cell.K;
+            if (AquiferHelpers::neighborCellInsideReservoirAndActive(grid, i, j, k, FaceDir::XPlus)) {
+                trans[0].insert(cell.global_index);
+            }
+            if (AquiferHelpers::neighborCellInsideReservoirAndActive(grid, i, j, k, FaceDir::XMinus)) {
+                trans[0].insert(grid.getGlobalIndex(i-1, j, k));
+            }
+            if (AquiferHelpers::neighborCellInsideReservoirAndActive(grid, i, j, k, FaceDir::YPlus)) {
+                trans[1].insert(cell.global_index);
+            }
+            if (AquiferHelpers::neighborCellInsideReservoirAndActive(grid, i, j, k, FaceDir::YMinus)) {
+                trans[1].insert(grid.getGlobalIndex(i, j-1, k));
+            }
+            if (AquiferHelpers::neighborCellInsideReservoirAndActive(grid, i, j, k, FaceDir::ZPlus)) {
+                trans[2].insert(cell.global_index);
+            }
+            if (AquiferHelpers::neighborCellInsideReservoirAndActive(grid, i, j, k, FaceDir::ZMinus)) {
+                trans[2].insert(grid.getGlobalIndex(i, j, k-1));
+            }
+        }
+        return trans;
+    }
+
+    void SingleNumericalAquifer::updateCellProps(const EclipseGrid& grid,
+                                                 std::vector<double>& pore_volume,
+                                                 std::vector<int>& satnum,
+                                                 std::vector<int>& pvtnum) const {
+        for (const auto& cell : this->cells_) {
+            const size_t activel_index =  grid.activeIndex(cell.global_index);
+            pore_volume[activel_index] = cell.poreVolume();
+            satnum[activel_index] = cell.sattable;
+            pvtnum[activel_index] = cell.pvttable;
+        }
+    }
+
+    void SingleNumericalAquifer::appendNNC(const EclipseGrid& grid, const FieldPropsManager &fp, NNC& nnc) const {
+        this->appendCellNNC(nnc);
+        this->appendConnectionNNC(grid, fp, nnc);
+    }
+
+    void SingleNumericalAquifer::appendCellNNC(NNC& nnc) const {
+        // aquifer cells are connected to each other through NNCs to form the aquifer
+        for (size_t i = 0; i < this->cells_.size() - 1; ++i) {
```

Comment:
The Aquifer object should not be allowed to mutate the NNC object. The AQUIFER object should just pass out a container of data (cell index + trans I guess) suitable to actually create the NNC's - and then either calling scope (EclipseState) - or the NNC implementation itself - should actually add the NNCs.

Reply:
I understand it. Similar to the FiledProps. I will address this.